hoomd/BoxResizeUpdater.cc

```diff
@@ -45,105 +49,193 @@ BoxResizeUpdater::~BoxResizeUpdater()
     m_exec_conf->msg->notice(5) << "Destroying BoxResizeUpdater" << endl;
     }
 
-/*! \param scale_particles Set to true to scale particles with the box. Set to false to leave particle positions alone
-    when scaling the box.
-*/
-void BoxResizeUpdater::setParams(bool scale_particles)
+void BoxResizeUpdater::setBox1(BoxDim box)
+    {
+    if (!m_py_box1.is_none())
+        {
+        m_py_box1 = m_py_box1.attr("__class__")
+                            .attr("_from_cpp")(box);
+        }
+    m_box1 = box;
+    }
+
+void BoxResizeUpdater::setBox1Py(pybind11::object box)
     {
-    m_scale_particles = scale_particles;
+    m_box1 = getBoxDimFromPyObject(box);
+    m_py_box1 = box;
     }
 
-/*! Perform the needed calculations to scale the box size
+void BoxResizeUpdater::setBox2(BoxDim box)
+    {
+    if (!m_py_box1.is_none())
+        {
+        m_py_box1 = m_py_box1.attr("__class__")
+                            .attr("_from_cpp")(box);
+        }
+    m_box2 = box;
+    }
+
+void BoxResizeUpdater::setBox2Py(pybind11::object box)
+    {
+    m_box2 = getBoxDimFromPyObject(box);
+    m_py_box2 = box;
+    }
+
+/// Get the current box based on the timestep
+BoxDim BoxResizeUpdater::getCurrentBox(unsigned int timestep)
+    {
+    Scalar min = m_variant->min();
+    Scalar max = m_variant->max();
+    Scalar cur_value = (*m_variant)(timestep);
+    Scalar scale = 0;
+    if (cur_value == max)
+        {
+        scale = 1;
+        }
+    else if (cur_value > min)
+        {
+        scale = (cur_value - min) / (max - min);
+        }
+
+    Scalar3 L1 = m_box1.getL();
+    Scalar3 L2 = m_box2.getL();
+    Scalar Lx = L2.x * scale + (1.0 - scale) * L1.x;
+    Scalar Ly = L2.y * scale + (1.0 - scale) * L1.y;
+    Scalar Lz = L2.z * scale + (1.0 - scale) * L1.z;
+    Scalar xy = m_box2.getTiltFactorXY() * scale +
+                (1.0 - scale) * m_box1.getTiltFactorXY();
+    Scalar xz = m_box2.getTiltFactorXZ() * scale +
+                (1.0 - scale) * m_box1.getTiltFactorXZ();
+    Scalar yz = m_box2.getTiltFactorYZ() * scale +
+                (1.0 - scale) * m_box1.getTiltFactorYZ();
+
+    BoxDim new_box = BoxDim(make_scalar3(Lx, Ly, Lz));
+    new_box.setTiltFactors(xy, xz, yz);
+    return new_box;
+    }
+
+/// Return whether two boxes are equivalent upto 1e-7
+bool BoxResizeUpdater::boxesAreEquivalent(BoxDim& box1, BoxDim& box2)
+    {
+    Scalar3 L1 = box1.getL();
+    Scalar3 L2 = box2.getL();
+
+    Scalar xy1 = box1.getTiltFactorXY();
+    Scalar xy2 = box2.getTiltFactorXY();
+    Scalar xz1 = box1.getTiltFactorXZ();
+    Scalar xz2 = box2.getTiltFactorXZ();
+    Scalar yz1 = box1.getTiltFactorYZ();
+    Scalar yz2 = box2.getTiltFactorYZ();
+
+    return fabs((L1.x - L2.x) / L1.x) < 1e-7 &&
```

Comment:
Yeah... I should have checked more closely. I copied and pasted the changes for each dimension, but must have missed where `L1` was used exclusively.

With regards to the zero check, that makes sense.

Reply:
Yes, I understand that the code was copied and pasted. I'm amazed at how long buggy code can survive before issues get noticed. 

This is used in a check to see if the current simulation box is identical to the one being set (i.e. in a flat part of a ramp after the ramp completes). In this case, there is no reason to perform the expensive box scale operation with a scale of 1.

Thinking about it again, the proper check for this would be `==` checks for all 6 box parameters. Maybe `BoxDim` should get an == operator instead of this method?

Reply:
I was referring to my own copying and pasting of my own changes (the typo was mine). Sorry if it came across as shifting the blame. I just meant that I was not careful enough with my own code changes.

Agreed on the second part.

I think the equality check would make sense.